File: JANUARY/06-01-2026/Max Xor Subarray of size K.java
class Solution {
    public int maxSubarrayXOR(int[] arr, int k) {
        int n = arr.length;
        if (k > n) return 0;
        int xorSum = 0;
        for (int i = 0; i < k; i++) xorSum ^= arr[i];
        int maxXor = xorSum;
        for (int i = k; i < n; i++) {
            xorSum ^= arr[i] ^ arr[i - k];
            maxXor = Math.max(maxXor, xorSum);
        }
        return maxXor;
    }
}

File: 2026_GFG_POTD/MARCH/04-03-2026/Max Xor Subarray of size K.py
class Solution:
    def maxSubarrayXOR(self, arr, k):
        n = len(arr)
        if k > n:
            return 0
        
        xor_sum = 0
        for i in range(k):
            xor_sum ^= arr[i]
        
        max_xor = xor_sum
        
        for i in range(k, n):
            xor_sum ^= arr[i] ^ arr[i - k]
            max_xor = max(max_xor, xor_sum)
        
        return max_xor